packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -2356,17 +2363,20 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin {
     if (paintCursorAboveText)
       _textPainter.paint(context.canvas, effectiveOffset);
 
-    if (showCaret) {
+    if (canShowCaret) {
       assert(selection != null);
-      _paintCaret(context.canvas, effectiveOffset, selection!.extent);
+      _tryPaintCaret(context.canvas, effectiveOffset, selection!.extent);
+      _updateCaretRect();
```

Comment:
We need the `cartRect` to bring `TextField` into view and scroll the long text although the `showCursor` is false.